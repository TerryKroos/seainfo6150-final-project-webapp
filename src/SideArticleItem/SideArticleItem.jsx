import React from 'react';
import styles from "./SideArticleItem.module.css";
import {BrowserRouter as Router, Link} from 'react-router-dom';
import { useHistory } from "react-router-dom";


const SideArticleItem = (props)=>{

    const history = useHistory();
    const handleSubmit = () => {
        history.push('/article', {params: props.article})
    }
    return(
      <div className={styles.container} onClick={handleSubmit}>
         <img className={styles.img}src={props.article.image} alt="article img"/>
         <div className={styles.articletext}>
           <Router>
           <Link className={styles.link}>
            {props.article.heading }</Link>
           </Router>
            
         </div>
      </div>
      )
}

export default SideArticleItem;