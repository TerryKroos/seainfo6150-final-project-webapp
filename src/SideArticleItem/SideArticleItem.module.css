.container {
    height: 70%;
    width: 30%;
    background-color: #F0F0F0;
    cursor: pointer;
    font-family: 'Inter', sans-serif;
    color: #556575;

    
   
}
.img {
 width: 100%;   
 height: auto;
 max-height: 170px;

}
.articletext {
    display: flex;
    justify-content: center;
    background-color: #F0F0F0;
    height: 4em;
    width:100%;
    font-size: 14px;
}
.link{
    text-decoration: none;
    color: #556575;
    cursor: pointer;
  }
  .link:hover{
    color: #282828; 
  }