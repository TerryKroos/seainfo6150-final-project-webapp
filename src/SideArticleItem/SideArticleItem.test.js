
import React from "react";
import SideArticleItem from "./SideArticleItem";
import { render } from "@testing-library/react";

describe("SideArticleItem tests", () => {
    it("renders correctly", () => {
      const { container } = render(<SideArticleItem article={{}}/>);
      expect(container).toMatchSnapshot();
    });
  
  });