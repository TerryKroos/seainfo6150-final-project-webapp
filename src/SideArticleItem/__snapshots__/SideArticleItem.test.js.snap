// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideArticleItem tests renders correctly 1`] = `
<div>
  <div
    class="container"
  >
    <img
      alt="article img"
      class="img"
    />
    <div
      class="articletext"
    >
      <a
        class="link"
        href=""
      />
    </div>
  </div>
</div>
`;
